plugins/inventory/hcloud.py

```diff
@@ -111,7 +111,7 @@ class InventoryModule(BaseInventoryPlugin, Constructable):
 
     def _configure_hcloud_client(self):
         self.token_env = self.get_option("token_env")
-        self.api_token = self.get_option("token") or os.getenv(self.token_env)
+        self.api_token = self.templar.template(self.get_option("token"), fail_on_undefined=True) or os.getenv(self.token_env)
```

Comment:
because of the `or os.getenv` shouldn't this be `fail_on_undefined=False`?

Reply:
Why not remove the `or os.getenv()` completely and let Ansible do that for you, by using `env:` with that option? (Like here: https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/inventory/yaml.py#L25)

Reply:
> because of the `or os.getenv` shouldn't this be `fail_on_undefined=False`?

Yes, that sounds reasonable. I'll fix it.